python/src/io.rs

```diff
@@ -284,13 +285,18 @@ pub struct PyIo(PyObject);
 impl Read for PyIo {
     fn read(&mut self, mut buf: &mut [u8]) -> io::Result<usize> {
         Python::with_gil(|py| {
+            if buf.is_empty() {
+                return Ok(0);
+            }
+            let to_read = max(1, buf.len() / 4); // We divide by 4 because TextIO works with number of characters and not with number of bytes
             let read = self
                 .0
-                .call_method(py, "read", (buf.len(),), None)
+                .as_ref(py)
+                .call_method1("read", (to_read,))
                 .map_err(to_io_err)?;
             let bytes = read
-                .extract::<&[u8]>(py)
-                .or_else(|e| read.extract::<&str>(py).map(str::as_bytes).map_err(|_| e))
+                .extract::<&[u8]>()
+                .or_else(|e| read.extract::<&str>().map(str::as_bytes).map_err(|_| e))
                 .map_err(to_io_err)?;
             buf.write_all(bytes)?;
```

Comment:
Not sure if this would really have helped with the issue, but this could actually be `.unwrap()` instead of `?` since it should never fail. (In this particular case, this would mean that users would have reported panic exceptions, i.e. `BaseException`s, instead of "random I/O errors". Not sure if this would have been nicer for your users, but it would certainly have been hard to ignore.)

Reply:
Good point. Thank you! Done.